opensearch/opensearch.yml.tpl

```diff
@@ -49,4 +65,35 @@ plugins.security.restapi.roles_enabled: ["all_access"]
 prometheus.metric_name.prefix: "opensearch_"
 prometheus.indices: true
 prometheus.cluster.settings: true
-prometheus.nodes.filter: "_all"
\ No newline at end of file
+prometheus.nodes.filter: "_all"
+
+%{ if try(opensearch_cluster.audit.enabled, false) ~}
+plugins.security.audit.type: ${opensearch_cluster.audit.storage_type}
+
+plugins.security.audit.config.enable_rest: true
+plugins.security.audit.config.enable_transport: true
+
+plugins.security.audit.config.index: ${opensearch_cluster.audit.index}
+plugins.security.audit.config.type: "auditlog"
+
+%{ if opensearch_cluster.audit.storage_type == "external_opensearch" && opensearch_cluster.audit.external != null ~}
+plugins.security.audit.config.http_endpoints: [${join(", ", [for e in opensearch_cluster.audit.external.http_endpoints: "\"${e}\""])}]
+
+plugins.security.audit.config.enable_ssl: true
+plugins.security.audit.config.verify_hostnames: true
+%{ if trimspace(try(opensearch_cluster.audit.external.auth.ca_cert, "")) != "" ~}
```

Comment:
I think you can just put a check against the empty string here (the default value if nothing is passed) to make it more concise.

I doubt very much that someone would define a value here that is just a space, a newline or something of the kind. I'm hopeful they would realize that passing those kinds of value for the ca cert is not a good idea.